server/boot/user.js

```diff
@@ -631,4 +645,51 @@ module.exports = function(app) {
       return res.render('account/forgot');
     });
   }
+
+  function getReportUserProfile(req, res) {
+    const username = req.params.username.toLowerCase();
+    return res.render('account/report-profile', {
+      title: 'Report User',
+      username
+    });
+  }
+
+  function postReportUserProfile(req, res) {
+    const username = req.params.username.toLowerCase();
+    const report = req.body.reportDescription;
+    const { user } = req;
+
+    return Email.send$({
+      type: 'email',
+      to: 'Team@FreeCodeCamp.com',
+      cc: user.email,
+      from: 'Team@FreeCodeCamp.com',
+      subject: 'Abuse Report : Reporting ' + username + '\'s profile.',
+      text: dedent(`
+        Hello Team,\n
+        This is to report the profile of ${username}.\n
+        Report Details:\n
+        ${report}\n\n
+        Reported by:
+        Username: ${user.username}
+        Name: ${user.name}
+        Email: ${user.email}\n
+        Thanks and regards,
+        ${user.name}
+      `)
+    }, err => {
+      if (err) {
+        debug(err);
+        req.flash('errors', {
+          msg: 'Oops, something went wrong, please try again later.'
+        });
+        return res.redirect('/' + user.username);
```

Comment:
This is redirect the reporting user to their own user profile. Is this intentional? shouldn't this send them back to the page they were on using req.params.username?